### ISSUE TITLE
sandbox.restore with args should throw

### ISSUE THREAD

**Issue by TheSavior** [sinonjs/sinon#1149]
* Sinon version: 1.17.5
* Environment: node / browser
* Example URL: https://runkit.com/5701d7f537a13d1200a55a1b/57ddaf833ffa1f1400543e75
* Other libraries you are using: none

I expected this code to throw since `sandbox.restore` doesn't take any arguments.

```js
const sinon = require("sinon")

const sandbox = sinon.sandbox.create();
sandbox.restore('foo', 'bar');
```

I've found that in a couple places in our code base, engineers think they can restore an individual stub by calling:
```js
sandbox.restore(stub);
```
Which actually restores **all** stubs

Whereas they should be calling:
```js
stub.restore();
```

I've been thinking of how I should address this problem, possibly by using a linter or by monkey patching the library. However, I figure that more than just our team has run into this and this is a thing sinon itself can protect engineers from doing.

**Comment by fatso83** [comment]
Sure, that seems like an obvious bug that's easy to fix. Could you perhaps create a PR to address this and I'll see to it that it gets merged?

**Comment by fatso83** [comment]
FYI: Do note that the fix in #1150 goes into `#master`, which is the upcoming Sinon 2.0, not the 1.17.x maintenance release. If you want it both places the patch needs to be reapplied against the 1.17 branch. 

To get the working pre-release, just change version by `npm install --save-dev sinon@next`.
